### PR TITLE
Refactor EnsureEchoService() for e2e tests to use a deployment

### PR DESCRIPTION
**Pull request by spencerhance**
EnsureEchoService() now creates a deployment instead of a single pod (existing tests use 1 replica).  This will be useful for e2e tests that test the creation of multiple NEGs.

Also includes some fixes from comments made on https://github.com/kubernetes/ingress-gce/pull/679, mainly refactoring some common checks into checkGCLB().

### CODE REVIEW

File: cmd/e2e-test/basic_test.go [hunk 1]
```diff
@@ -101,12 +100,9 @@ func TestBasic(t *testing.T) {
 				t.Fatalf("Error getting GCP resources for LB with IP = %q: %v", vip, err)
 			}
 
-			// Do some cursory checks on the GCP objects.
-			if len(gclb.ForwardingRule) != tc.numForwardingRules {
-				t.Errorf("got %d fowarding rules, want %d;\n%s", len(gclb.ForwardingRule), tc.numForwardingRules, pretty.Sprint(gclb.ForwardingRule))
-			}
-			if len(gclb.BackendService) != tc.numBackendServices {
-				t.Errorf("got %d backend services, want %d;\n%s", len(gclb.BackendService), tc.numBackendServices, pretty.Sprint(gclb.BackendService))
+			err = e2e.CheckGCLB(gclb, tc.numForwardingRules, tc.numBackendServices)
```
Comment (MrHohn) [inline cmd/e2e-test/basic_test.go]: nit (and below):
```golang
if err := e2e.CheckGCLB(gclb, tc.numForwardingRules, tc.numBackendServices); err != nil {
    t.Error(err)
}
```

**Comment by MrHohn** [comment]
Thanks for squashing the commits :)
/lgtm

File: pkg/e2e/fixtures.go [hunk 2]
```diff
@@ -76,10 +75,21 @@ func EnsureEchoService(s *Sandbox, name string, annotations map[string]string,
...
+			Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"app": name}},
+			Template: podTemplate,
+		},
+	}
+
 	svc, err := s.f.Clientset.CoreV1().Services(s.Namespace).Get(name, metav1.GetOptions{})
 
 	if svc == nil || err != nil {
-		if pod, err = s.f.Clientset.Core().Pods(s.Namespace).Create(pod); err != nil {
+		if deployment, err = s.f.Clientset.Extensions().Deployments(s.Namespace).Create(deployment); err != nil {
```
[the remark this answers is not in the document]
Comment (spencerhance) [reply pkg/e2e/fixtures.go]: Good call.  Since this PR got merged I can add this to another one.

[the remark this answers is not in the document]
Comment (spencerhance) [reply pkg/e2e/fixtures.go]: https://github.com/kubernetes/ingress-gce/pull/702

**Pull request merged**